File: src-tauri/src/elements/value_objects/meta.rs
use chrono::{DateTime, Utc};
use fractional_index::FractionalIndex;
use uuid::Uuid;

use crate::elements::value_objects::element_id::ElementId;

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Meta {
    pub element_id: ElementId,
    pub name: String,
    pub parent: Option<ElementId>,
    pub position: FractionalIndex,
    pub study_profile_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub modified_at: DateTime<Utc>,
}

File: src/types/elements/meta.ts
import { ElementId } from "./elementId";
import { Tag } from "./tag";

export interface Meta {
	elementId: string;
	name: string;
	parent: ElementId | null;
	position: number;
	tags: Tag[];
	createdAt: string;
	modifiedAt: string;
	removedAt: string | null;
}
